internal/characters/faruzan/asc.go

```diff
@@ -0,0 +1,53 @@
+package faruzan
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+)
+
+const (
+	a4Key    = "faruzan-a4"
+	a4ICDKey = "faruzan-a4-icd"
+)
+
+func (c *char) a4() {
+	count := 0
+	c.Core.Events.Subscribe(event.OnEnemyHit, func(args ...interface{}) bool {
+		atk := args[1].(*combat.AttackEvent)
+		if atk.Info.Element != attributes.Anemo {
+			return false
+		}
+
+		switch atk.Info.AttackTag {
+		case combat.AttackTagNormal,
+			combat.AttackTagExtra,
+			combat.AttackTagPlunge,
+			combat.AttackTagElementalArt,
+			combat.AttackTagElementalBurst:
+		default:
+			return false
+		}
+
+		char := c.Core.Player.ByIndex(atk.Info.ActorIndex)
+		if char.StatusIsActive(burstBuffKey) && !char.StatusIsActive(a4ICDKey) {
+			char.AddStatus(a4Key, 6, false)
```

Comment:
both of these should only apply to Faruzan (i.e. A4 icd is shared between characters), so please rework A4 to reflect this. In that case, you can drop `count`.